homeassistant/components/wallbox/number.py

```diff
@@ -58,27 +66,33 @@ class WallboxNumber(CoordinatorEntity, NumberEntity):
     """Representation of the Wallbox portal."""
 
     entity_description: WallboxNumberEntityDescription
+    coordinator: WallboxCoordinator
 
     def __init__(
-        self, coordinator, config, description: WallboxNumberEntityDescription
-    ):
+        self,
+        coordinator: WallboxCoordinator,
+        entry: ConfigEntry,
+        description: WallboxNumberEntityDescription,
+    ) -> None:
         """Initialize a Wallbox sensor."""
         super().__init__(coordinator)
         self.entity_description = description
         self._coordinator = coordinator
-        self._attr_name = f"{config.title} {description.name}"
+        self._attr_name = f"{entry.title} {description.name}"
         self._attr_min_value = description.min_value
 
     @property
-    def max_value(self):
+    def max_value(self) -> float:
         """Return the maximum available current."""
-        return self._coordinator.data[CONF_MAX_AVAILABLE_POWER_KEY]
+        return cast(float, self._coordinator.data[CONF_MAX_AVAILABLE_POWER_KEY])
 
     @property
-    def value(self):
+    def value(self) -> float | None:
         """Return the state of the sensor."""
-        return self._coordinator.data[CONF_MAX_CHARGING_CURRENT_KEY]
+        return cast(
+            Optional[float], self._coordinator.data[CONF_MAX_CHARGING_CURRENT_KEY]
+        )
 
-    async def async_set_value(self, value: float):
+    async def async_set_value(self, value: float) -> None:
         """Set the value of the entity."""
         await self._coordinator.async_set_charging_current(value)
```

Comment:
See comments in __init__, you are sending a float to a function which accepts an int, is this intentional?